### ISSUE TITLE
Incorrect '-I' option for non-packed libraries

### ISSUE THREAD

**Issue by pveber** [solvuu/solvuu-build#30]
The code (see [here for libs](https://github.com/solvuu/solvuu-build/blob/master/lib/project.ml#L664) and [there for apps](https://github.com/solvuu/solvuu-build/blob/master/lib/project.ml#L837)) generating `-I` options when something depends on an internal library is incorrect in the case of non-packed libraries. The use of `dirname` works only for packed libraries, otherwise one should keep the lib directory.

A reproducible case is available on the [solvuu-build-I-bug](https://github.com/pveber/pipes/tree/solvuu-build-I-bug) branch of my `pipes` library.

**Comment by agarwal** [comment]
Unfortunately links to specific line numbers get out of sync as new commits are pushed. Stable GitHub links should include the commit ID, e.g. I think these are the corrected links:

- [here for libs](https://github.com/solvuu/solvuu-build/blob/a601eb57b21418de34ed209bc129fae887386974/lib/project.ml#L664)
- [there for apps](https://github.com/solvuu/solvuu-build/blob/a601eb57b21418de34ed209bc129fae887386974/lib/project.ml#L837)

More importantly, I don't immediately see why it is wrong currently. I'm checking out your `pipes` repo and investigating further.

**Comment by agarwal** [comment]
The `solvuu-build-I-bug` doesn't merge with `master` anymore. Maybe you can please fix it up, and I'll try again.

**Comment by pveber** [comment]
Fortunately there is no need to merge with `master` to observe the issue, you can just check it out and try to compile. Normally, you should observe the following error:
```
+ ocamlfind ocamlc -annot -bin-annot -c -for-pack Pipes_benchmark -g -I lib -I lib/benchmark -o lib/benchmark/int_sum.cmo -safe-string -short-paths -thread -w A-4-33-41-42-44-45-48 -package cfstream,core_bench lib/benchmark/int_sum.ml
File "lib/benchmark/int_sum.ml", line 20, characters 11-15:
Error: Unbound module Pipe
```
The compiler should have received a `-I lib/pure` option but here it doesn't and as a result the module `Pipe` is unknown.

**Comment by agarwal** [comment]
First, note your `pure` lib is defined with the `Basic` style, while `benchmark` is `Pack`. I suspect this is unintended. Probably you want `pure`'s definition to call `make_lib` instead of `Project.lib`. This should resolve your immediate problem.

However, what you did should technically work. I'll leave this issue open to fix the `-I` dirs. I'll fix this.

**Comment by pveber** [comment]
Actually it is intended, I wanted to have a library `pipe` consisting of a single module `Pipe` and other packed libraries named `Pipe_foo`.

I agree with your description of the problem (and sorry for the outdated links).

**Comment by agarwal** [comment]
> Actually it is intended

Ah okay. So in that case, I'll prioritize fixing this. Shouldn't be too hard.

**Comment by agarwal** [comment]
Fixed in 3de5811856082ee6a55de40bb1ccc3037e235ecb.
